upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js

```diff
@@ -36,16 +37,19 @@ function callEndpoint(path, postBody, callback) {
 
     const req = https.request(options, (res) => {
         var chunks = [];
+        var bodyString = '';
         
         res.on("data", function (chunk) {
             chunks.push(chunk);
+            bodyString += chunk.toString()
         });
 
        res.on('end', function() {
            if (callback) {
+               var responseMessage = res.statusCode == 418 ? bodyString : res.statusMessage;
```

Comment:
I'm not clear what is going on here; a comment might be nice. I don't understand how it wasn't returning JSON before. Is it because bodyString is stringified JSON in this case, and it's not when the error code is not 418?

If so, seems like there could be a more generic test to see whether bodyString is JSON or not, just in case there are other error codes that also have JSON responses.

Probably overkill, just random thoughts :)

Reply:
So the issue is that the way we are returning CustomWebApplicationExceptions is different than what the library typically expects. We are simply returning a TEXT-PLAIN with the error message instead of a JSON of a form like {"statusCode": ..., "statusMessage": ...}

Technically this happens for any time we throw CustomWebApplicationException. I agree with your general approach, I can try that out. Since we do throw more than just 418 errors with CustomWebApplicationException.